### ISSUE TITLE
New Rule: ban-eslint-disable

### ISSUE THREAD

**Issue by septs** [eslint/eslint#15534]
### Rule details

Require provide reason for disable eslint rule

### Related ECMAScript feature

N/A

### What type of rule is this?

Enforces a formatting/stylistic preference

### Example code

```js
// valid example
// eslint-disable-next-line no-console -- Log a error
console.error(error)

// invalid example
// eslint-disable-next-line
```

### Why should this rule be in the core instead of a plugin?

Require provide reason for disable eslint rule

### Participation

- [ ] I am willing to submit a pull request to implement this rule.

### Additional comments

Port of https://typescript-eslint.io/rules/ban-ts-comment/

**Comment by aladdin-add** [comment]
you can achieve it using [eslint-plugin-eslint-comments](https://mysticatea.github.io/eslint-plugin-eslint-comments/)

**Comment by nzakas** [comment]
Thanks for the submission. Unfortunately, this rule doesn't meet our [criteria for new rules](https://eslint.org/blog/2020/05/changes-to-rules-policies), so I'm closing this issue. Please see the referenced plugin.
